lib/awful/prompt.lua

```diff
@@ -461,17 +461,18 @@ function prompt.run(args, textbox, exe_callback, completion_callback,
                     for _,v2 in ipairs(v[1]) do
                         match = match and mod[v2]
                     end
-                    if match or #filtered_modifiers == 0 then
+                    if match then
```

Comment:
Is the `or`-part removed on purpose? If so, I don't see why.

The rest looks good to me.